hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -98,18 +98,19 @@ protected void handlePutKeyEvent(OmKeyInfo keyInfo, Map<Long,
       nsSummary = new NSSummary();
     }
     int[] fileBucket = nsSummary.getFileSizeBucket();
-    
-    // Update immediate parent's totals (these fields now represent totals)
+
+    // Update immediate parent's totals (includes all descendant files)
     nsSummary.setNumOfFiles(nsSummary.getNumOfFiles() + 1);
     nsSummary.setSizeOfFiles(nsSummary.getSizeOfFiles() + keyInfo.getDataSize());
+    nsSummary.setReplicatedSizeOfFiles(nsSummary.getReplicatedSizeOfFiles() + keyInfo.getReplicatedSize());
```

Comment:
```suggestion
    // Before arithmetic operations, check for sentinel value
  long currentReplSize = nsSummary.getReplicatedSizeOfFiles();
  if (currentReplSize < 0) {
    // Old data, initialize to 0 before first use
    currentReplSize = 0;
    nsSummary.setReplicatedSizeOfFiles(0);
  }
  nsSummary.setReplicatedSizeOfFiles(currentReplSize + keyInfo.getReplicatedSize());
```
Check similar data corruption issues else where in code.

Reply:
Addressed.